scripts/ASTImportTest.sh

```diff
@@ -14,90 +45,91 @@ SOLIDITY_BUILD_DIR=${SOLIDITY_BUILD_DIR:-${REPO_ROOT}/build}
 SOLC=${SOLIDITY_BUILD_DIR}/solc/solc
 SPLITSOURCES=${REPO_ROOT}/scripts/splitSources.py
 
+# shellcheck source=scripts/common.sh
+source "${REPO_ROOT}/scripts/common.sh"
+
 SYNTAXTESTS_DIR="${REPO_ROOT}/test/libsolidity/syntaxTests"
 ASTJSONTESTS_DIR="${REPO_ROOT}/test/libsolidity/ASTJSON"
-NSOURCES="$(find "$SYNTAXTESTS_DIR" -type f | wc -l)"
-
-# DEV_DIR="${REPO_ROOT}/../tmp/contracts/"
-# NSOURCES="$(find $DEV_DIR -type f | wc -l)" #TODO use find command
 
 FAILED=0
 UNCOMPILABLE=0
 TESTED=0
 
 if [[ "$(find . -maxdepth 0 -type d -empty)" == "" ]]; then
-    echo "Test directory not empty. Skipping!"
-    exit 1
+    fail "Test directory not empty. Skipping!"
 fi
 
-# function tests whether exporting and importing again leaves the JSON ast unchanged
+function ast_import_export_equivalence
+{
+    local sol_file="$1"
+    local input_files="$2"
+    # save exported json as expected result (silently)
+    $SOLC --combined-json ast --pretty-json --json-indent 4 "${input_files}" > expected.json 2> ${REDIRECT_TO}
+    # import it, and export it again as obtained result (silently)
+    if ! $SOLC --import-ast --combined-json ast --pretty-json --json-indent 4 expected.json > obtained.json 2> stderr.txt
+    then
+        # For investigating, use exit 1 here so the script stops at the
+        # first failing test
+        # exit 1
+        FAILED=$((FAILED + 1))
+        printError "ERROR: AST reimport failed for input file $sol_file"
+        printError
+        printError "Compiler stderr:"
+        cat ./stderr.txt >&2
+        printError
+        printError "Compiler stdout:"
+        cat ./obtained.json >&2
+        return 1
+    fi
+    if ! diff_files expected.json obtained.json
+    then
+        FAILED=$((FAILED + 1))
+    fi
+    TESTED=$((TESTED + 1))
+    rm expected.json obtained.json
```

Comment:
Why this is not forced like the one below? Is it expected that the user confirm the removal of those files?

Reply:
nope, I changed it now to be forced